types/types.cc

```diff
@@ -89,18 +89,15 @@ time_point_to_string(const T& tp)
     };
 
     auto millis = d.rem;
-    if (!millis) {
-        return fmt::format("{}", to_string(tm));
-    }
     // adjust seconds for time points earlier than posix epoch
     // to keep the fractional millis positive
     if (millis < 0) {
         millis += 1000;
         seconds--;
         gmtime_r(&seconds, &tm);
     }
-    auto micros = millis * 1000;
-    return fmt::format("{}.{:06d}", to_string(tm), micros);
+    millis = std::round(millis * 1000) / 1000;
```

Comment:
what is this for?

Reply:
to round to 3 decimal places

Reply:
`millis` is a `long int`, so if we multiply it by 1000 if it does not overflow, the result would be `millis * 1000`. after dividing it by `1000`, the result should be `millis`. am i missing anything? or could you please raise an example?

Reply:
oops, my mistake, will fix
thanks